upstairs/src/dummy_downstairs_tests.rs

```diff
@@ -2988,21 +2988,31 @@ async fn test_ro_activate_from_list(activate: [bool; 3]) {
         guest,
     };
 
+    // Get downstairs state before we send IO.
+    // If a downstairs is not present before we send IO, then we won't bother
+    // calling ack_read() for that downstairs.  It's possible the downstairs
+    // joins just after  we check, and that is okay for this test.  The race
+    // here is when a downstairs should join, and eventually will but has not
+    // joined quite yet and the IO for that downstairs is skipped, then we go
+    // on an expect it to have arrived and fail the ack_read() below.
+    let ds = harness.guest.downstairs_state().await.unwrap();
+
```

Comment:
Is there not still a race condition here?  What if the Downstairs is not active when we call `downstairs_state`, but becomes active before spawning the read?

Reply:
Its is still a race, but if we lose this path (it becomes active afterwards) it just means we won't check for and respond to the read on that downstairs.

We must have at least one downstairs that has gone active because we `guest.query_is_active()`, so that one will be responsible for sending the ack.

Hmm. 
We could update the test to wait for as many downstairs as are "true" before proceeding with the read, that would avoid hitting the situation where we activate before the 2nd downstairs has joined.

Reply:
>  We could update the test to wait for as many downstairs as are "true" before proceeding with the read, that would avoid hitting the situation where we activate before the 2nd downstairs has joined.

Yeah, I think it makes sense to wait until all of the expected Downstairs are active before sending the read – that way, everything is in a known state.